src/v/cluster/data_migration_worker.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "base/outcome.h"
+#include "cluster/data_migration_types.h"
+#include "cluster/fwd.h"
+#include "errc.h"
+#include "model/fundamental.h"
+
+#include <seastar/core/sharded.hh>
+#include <seastar/core/shared_ptr.hh>
+
+namespace cluster::data_migrations {
+
+class worker : public ss::peering_sharded_service<worker> {
+public:
+    worker(
+      model::node_id self,
+      ss::sharded<partition_leaders_table>&,
+      ss::sharded<ss::abort_source>&);
+
+    ss::future<errc> perform_partition_work(
+      model::ntp ntp, id migration, state sought_state, bool _tmp_wait_forever);
+    void abort_partition_work(model::ntp ntp);
```

Comment:
can `ntp` be passed as a `const` ref ?

Reply:
Regardless how an argument is passed to `invoke_on`, when it calls the function it passes args as rvalue references anyway. It is not likely to change as far as I understand: to call a function on another core Seastar has to copy or move the argument to that core; it does not need them after the call, so it makes perfect sense to surrender them.

I can rather make it an rvalue ref here for visibility.